### PR TITLE
Fixes to camera management

### PR DESCRIPTION
**Pull request by IJDykeman**

### CODE REVIEW

File: config/hoverjet/camera.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ auto_focus: 0
 auto_exposure: 0.25
 # Set a fixed exposure (The minimum)
 webcam_exposure: 0.1
+exposure: 0.1
 
 # Calling our shots: The camera we expect to be running
-serial_number: 5CFD076E
+serial_number: CA38DB5E
```
Comment (benkroop) [inline config/hoverjet/camera.yaml]: Do we actually want to change the camera serial number of master?

File: camera/camera_manager.cc [hunk 2]
```diff
@@ -43,12 +43,19 @@ std::optional<YAML::Node> CameraManager::read_YAML(const std::string& filepath)
...
+  std::cout << "  " << std::endl;
   camera.serial_number = cfg["serial_number"].as<std::string>();
   camera.v4l_path = "/dev/v4l/by-id/" + cfg["v4l_path"].as<std::string>();
-  camera.video_index = *follow_v4l_path(camera.v4l_path);
-  if (!camera.video_index) {
+  const auto optional_camera_index = follow_v4l_path(camera.v4l_path);
+  
+  if (!optional_camera_index.has_value()){
+    std::cout << "Could not get video index from v4l." << std::endl;
+    std::cout << "  " << cfg["v4l_path"].as<std::string>() << std::endl;
```
Comment (benkroop) [inline camera/camera_manager.cc]: Remove debug output.

File: camera/camera_manager.cc [hunk 3]
```diff
@@ -43,12 +43,19 @@ std::optional<YAML::Node> CameraManager::read_YAML(const std::string& filepath)
 std::optional<Camera> CameraManager::parse_config(const YAML::Node& cfg) const {
   Camera camera;
   // path & video index
+  std::cout << "  " << std::endl;
```
Comment (benkroop) [inline camera/camera_manager.cc]: Remove debug output

File: camera/camera_balsaq.cc [hunk 4]
```diff
@@ -44,7 +44,7 @@ void CameraBq::init(const Config& config) {
   cap = cv::VideoCapture(camera.video_index);
 
   camera_config_ = generate_capture_config(config);
-
+  std::cout << "returned camera config to camera bq" << std::endl;
```
Comment (benkroop) [inline camera/camera_balsaq.cc]: Remove debug output

File: camera/camera_manager.cc [hunk 5]
```diff
@@ -60,8 +67,13 @@ std::optional<Camera> CameraManager::parse_config(const YAML::Node& cfg) const {
 std::optional<int> CameraManager::follow_v4l_path(const std::string& path) const {
   std::error_code ec;
   const std::string v4l_path = fs::canonical(path, ec).string();
+  std::cout << "   input path:" << path << std::endl;
```
Comment (benkroop) [inline camera/camera_manager.cc]: Remove debug output

File: camera/camera_manager.cc [hunk 6]
```diff
@@ -45,10 +45,14 @@ std::optional<Camera> CameraManager::parse_config(const YAML::Node& cfg) const {
   // path & video index
   camera.serial_number = cfg["serial_number"].as<std::string>();
   camera.v4l_path = "/dev/v4l/by-id/" + cfg["v4l_path"].as<std::string>();
-  camera.video_index = *follow_v4l_path(camera.v4l_path);
-  if (!camera.video_index) {
+  const auto optional_camera_index = follow_v4l_path(camera.v4l_path);
+  
+  if (!optional_camera_index.has_value()){
```
Comment (jpanikulam) [inline camera/camera_manager.cc]: You can do `if(optional)` in C++, no need to do has_value

File: camera/camera_manager.cc [hunk 7]
```diff
@@ -45,10 +45,14 @@ std::optional<Camera> CameraManager::parse_config(const YAML::Node& cfg) const {
...
     return std::nullopt;
   }
+  else{
+    camera.video_index = optional_camera_index.value();
```
Comment (jpanikulam) [inline camera/camera_manager.cc]: prefer `*optional_camera_index` instead of `.value()`

**Pull request merged**